arangod/Agency/Agent.cpp

```diff
@@ -1775,15 +1775,18 @@ query_t Agent::gossip(query_t const& in, bool isCallback, size_t version) {
   }
     
   std::string err;
+  bool changed = false;
     
   /// Pool incomplete or the other guy is in my pool: I'll gossip.
   if (!_config.poolComplete() || _config.matchPeer(id, endpoint)) {
-      
+
+    auto before = _config.version();
     if (!_config.upsertPool(pslice, id)) {
       LOG_TOPIC(FATAL, Logger::AGENCY) << "Discrepancy in agent pool!";
       FATAL_ERROR_EXIT();      /// disagreement over pool membership are fatal!
     }
     auto pool = _config.pool();
+    changed = before < _config.version();
```

Comment:
What causes _config.version() to change between line 1783 and line 1789?

Reply:
_config.upsertPool(pslice, id)

Reply:
This is actually, what prevents the craziness now going forward.